routes/api/v1/foods.js

```diff
@@ -1,40 +1,91 @@
 var express = require("express");
 var router = express.Router();
 var Food = require('../../../models').Food;
-
-var pry = require('pryjs');
 const fetch = require('node-fetch');
+var pry = require('pryjs');
 
 router.get("/", function(req, res, next) {
   res.setHeader("Content-Type", "application/json");
-
   Food.findAll()
-  .then(foods => {
-    res.status(200).send(JSON.stringify(foods));
-  })
-  .catch(error => {
-    res.status(500).send({error})
-  });
+    .then(foods => {
+      res.status(200).send(JSON.stringify(foods));
+    })
+    .catch(error => {
+      res.status(500).send({
+        error
+      })
+    });
 });
 
 router.get("/:id", function(req, res, next) {
   res.setHeader("Content-Type", "application/json");
-
   Food.findOne({
-    where: {
-      id: req.params.id
-    }
-  })
-  .then(food => {
-    if(!food){
-      res.status(404).send({ error: "Requested food item could not be found." });
-    }else{
-      res.status(200).send(JSON.stringify(food));
-    }
-  })
-  .catch(error => {
-    res.status(500).send({error})
-  });
+      where: {
+        id: req.params.id
+      }
+    })
+    .then(food => {
+      if (!food) {
+        res.status(404).send({
+          error: "Requested food item could not be found."
+        });
+      } else {
+        res.status(200).send(JSON.stringify(food));
+      }
+    })
+    .catch(error => {
+      res.status(500).send({
+        error
+      })
+    });
 });
 
+router.post("/", function(req, res, next) {
+  res.setHeader("Content-Type", "application/json");
+  const calOrNameErrorMessage = "Name/Calories must be passed in to the body via x-www-form-urlencoded in the format of name or calories as the key and item name or calories count as the value without quotes"
+  if (req.body.name === undefined) {
+    res.status(400).send(JSON.stringify(calOrNameErrorMessage))
+  } else {
+    const name = upCase(req.body.name)
+    Food.findOrCreate({
```

Comment:
This is currently sending back a food object if the intended object already exists. We discussed sending a error back instead of the exisiting food object. I will work on this with a different PR.